File: onlinejudge/leetcode/1352.java
class ProductOfNumbers {

    private List<Integer> arr = new ArrayList<>();
    private int prod = 1;

    public void add(int num) {
        if (num == 0) {
            arr = new ArrayList<>();
            prod = 1;
            return;
        }
        prod *= num;
        arr.add(prod);
    }
    
    public int getProduct(int k) {
        if (arr.size() < k) {
            return 0;
        }

        int answer = arr.get(arr.size() - 1);
        if (arr.size() == k) {
            return answer;
        }
        return answer / arr.get(arr.size() - 1 - k);
    }
}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.add(num);
 * int param_2 = obj.getProduct(k);
 */

File: onlinejudge/leetcode/1352.rb
class ProductOfNumbers
  def initialize()
      @arr = []
      @prod = 1
  end


=begin
  :type num: Integer
  :rtype: Void
=end
  def add(num)
  if num == 0
      @arr = []
      @prod = 1
      return
  end
  @prod *= num
  @arr << @prod
  end


=begin
  :type k: Integer
  :rtype: Integer
=end
  def get_product(k)
      return 0 if @arr.size < k

      answer = @arr[-1]
      return answer if @arr.size == k
      
      answer / @arr[-1 - k]
  end

end

# Your ProductOfNumbers object will be instantiated and called as such:
# obj = ProductOfNumbers.new()
# obj.add(num)
# param_2 = obj.get_product(k)
